generator/src/test/java/com/scottlogic/deg/generator/Guice/ProfileProviderTests.java

```diff
@@ -0,0 +1,50 @@
+package com.scottlogic.deg.generator.Guice;
+
+import com.scottlogic.deg.generator.Profile;
+import com.scottlogic.deg.generator.generation.GenerationConfigSource;
+import com.scottlogic.deg.generator.inputs.InvalidProfileException;
+import com.scottlogic.deg.generator.inputs.JsonProfileReader;
+import org.junit.jupiter.api.Test;
+
+import java.io.File;
+import java.io.IOException;
+
+import static org.mockito.Mockito.*;
+
+public class ProfileProviderTests {
+
+    private GenerationConfigSource configSource = mock(GenerationConfigSource.class);
+    private JsonProfileReader profileReader = mock(JsonProfileReader.class);
+    private Profile profile = mock(Profile.class);
+    private File file = mock(File.class);
+    private ProfileProvider profileProvider = new ProfileProvider(configSource, profileReader);
+
+    @Test
+    public void get_profileIsNull_profileReadIsCalled() throws IOException, InvalidProfileException {
+        //Arrange
+        profileProvider.setProfile(null);
```

Comment:
`when(profileReader.read()).the return(null);`

This would remove the need for `setProfile()`